File: frontend/lib/models/product.dart
class Product {
  final int productId;
  final String name;
  final String description;
  final double price;
  final int currentStock;
  final String stockStatus;

  Product({
    required this.productId,
    required this.name,
    required this.description,
    required this.price,
    required this.currentStock,
    required this.stockStatus,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      productId: json['product_id'],
      name: json['name'],
      description: json['description'] ?? '',
      price: json['price'].toDouble(),
      currentStock: json['current_stock'] ?? 0,
      stockStatus: json['stock_status'] ?? 'Unknown',
    );
  }
}

File: backend/app/schemas/product.py
from pydantic import BaseModel, Field, field_validator
from typing import Optional
from decimal import Decimal

class ProductBase(BaseModel):
    name: str = Field(..., min_length=1, max_length=100)
    description: Optional[str] = Field(None, max_length=500)
    price: Decimal = Field(..., gt=0, decimal_places=2)

class ProductCreate(ProductBase):
    pass

class ProductUpdate(BaseModel):
    name: Optional[str] = Field(None, min_length=1, max_length=100)
    description: Optional[str] = None
    price: Optional[Decimal] = Field(None, gt=0, decimal_places=2)

    @field_validator('price')
    def validate_price(cls, v):
        if v is not None and v <= 0:
            raise ValueError('Price must be greater than 0')
        return v

class ProductResponse(ProductBase):
    product_id: int
    current_stock: Optional[int] = 0
    stock_status: Optional[str] = "Unknown"

    class Config:
        from_attributes = True